### DISCUSSION TITLE
Add Open/Load Form function with Async

### DISCUSSION THREAD

**Discussion by YogiYang** [twinbasic/twinbasic#462] [Ideas]
Hello,

The subject is quite cryptic so let me try and explain.

As per VB6 standards we can load and show a form and if we show the form as vbModal then the code after the show line will only get executed after the form closes. This is a norm in almost all languages and we generally are used to it.

Now if we load and show a form as non vbModal the code after the show line will get executed after the form is show.

I think we should add a Form Show feature called vbModalAsync. Here the developer has to provide a call back function which will be called when the Form is closed/unloaded.

I think if such a feature is implemented it would have multiple applications.

@WaynePhillipsEA please give your views on this.

**Comment by dmrvb** [comment]
You can put any code you want in the forms unload event so can call a specified sub of your choosing. Your request is really about creating callbacks where the calling code specifies the callback sub?. I think you can sort of do this with API calls but it would be good if TwinBasic made it batter and easier.
You can already do a good approximation to a callback in VB. The callback sub needs to have a fixed name but you can specify the form (or object) where that sub is located
